ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CommandTracker.scala

```diff
@@ -197,33 +209,40 @@ private[commands] class CommandTracker[Context](
 
       import CommandTracker.nonTerminalCodes
 
-      private def handleSubmitResponse(submitResponse: Ctx[(Context, String), Try[Empty]]) = {
-        val Ctx((_, commandId), value, _) = submitResponse
+      private def handleSubmitResponse(
+          submitResponse: Ctx[(Context, TrackedCommandKey), Try[Empty]]
+      ) = {
+        val Ctx((_, commandKey), value, _) = submitResponse
         value match {
           case Failure(GrpcException(status @ GrpcStatus(code, _), metadata))
               if !nonTerminalCodes(code) =>
-            getOutputForTerminalStatusCode(commandId, GrpcStatus.toProto(status, metadata))
+            getOutputForTerminalStatusCode(commandKey, GrpcStatus.toProto(status, metadata))
           case Failure(throwable) =>
             logger.warn(
               s"Service responded with error for submitting command with context ${submitResponse.context}. Status of command is unknown. watching for completion...",
               throwable,
             )
             None
           case Success(_) =>
-            logger.trace("Received confirmation that command {} was accepted.", commandId)
+            logger.trace(
+              "Received confirmation that command {} from submission {} was accepted.",
+              commandKey.commandId,
+              commandKey.submissionId,
+            )
             None
         }
       }
 
       @nowarn("msg=deprecated")
       private def registerSubmission(submission: Ctx[Context, CommandSubmission]): Unit = {
         val commands = submission.value.commands
+        val submissionId = commands.submissionId
```

Comment:
The `CommandService`should set the submission ID to a UUID if the application has left it empty in its submission. I haven't seen any such change in this PR. Will this be done separately?

Reply:
That is already the case, being done here https://github.com/digital-asset/daml/blob/main/ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/CommandsValidator.scala#L188

Reply:
It is being done in `CommandsValidator.validateCommands` which is being run by `GrpcCommandService` earlier. It was done as part of [KVL-999](https://digitalasset.atlassian.net/browse/KVL-999).

Reply:
Great. Good to know. Do we have test cases for that?

Reply:
Hmm, sadly, I don't see any. Will have to add them as part of this story.

Reply:
Followed up in: https://github.com/digital-asset/daml/pull/10882